js/views/reader_view.js

```diff
@@ -519,58 +519,49 @@ ReadiumSDK.Views.ReaderView = function(options) {
      * @fires ReadiumSDK.Events.SETTINGS_APPLIED
      */
     this.updateSettings = function(settingsData) {
-
-//console.debug("UpdateSettings: " + JSON.stringify(settingsData));
-
-        _viewerSettings.update(settingsData);
-        
-        if (_mediaOverlayPlayer)
-        {
-            _mediaOverlayPlayer.setAutomaticNextSmil(_viewerSettings.mediaOverlaysAutomaticPageTurn ? true : false);
-        }
-        
-        if(_currentView && !settingsData.doNotUpdateView) {
-
-            var bookMark = _currentView.bookmarkCurrentPage();
-
-            if(bookMark && bookMark.idref) {
-     
-                var wasPlaying = false;
-                if (_currentView.isReflowable && _currentView.isReflowable())
-                {
-                    wasPlaying = self.isPlayingMediaOverlay();
-                    if (wasPlaying)
-                    {
-                        self.pauseMediaOverlay();
-                    }
-                }
-
-                var spineItem = _spine.getItemById(bookMark.idref);
-                
-                initViewForItem(spineItem, function(isViewChanged){
-
-                    if(!isViewChanged) {
-                        _currentView.setViewSettings(_viewerSettings);
-                    }
-
-                    self.openSpineItemElementCfi(bookMark.idref, bookMark.contentCFI, self);
-
-                    if (wasPlaying)
-                    {
-                        self.playMediaOverlay();
-                        // setTimeout(function()
-                        // {
-                        // }, 60);
-                    }
-
-                    self.trigger(ReadiumSDK.Events.SETTINGS_APPLIED);
-                    return;
```

Comment:
Good catch on this bug, although the original coding error ; if I am not mistaken ; is that the `return` statement is misplaced (line 567 should be shifted down, after the `initViewForItem()` call). I think that this would be a cleaner fix than duplicating the `SETTINGS_APPLIED` event trigger for each `if then else` conditional block. What do you think?